lib/hd/common.js

```diff
@@ -86,6 +92,10 @@ common.parsePath = function parsePath(path, hard) {
     if ((index >>> 0) !== index)
       throw new Error('Path index out of range.');
 
+    if (index > 0x7fffffff) {
+      throw new Error(`Key path value ${index} is out of range`);
+    }
```

Comment:
When parsing a key path from a descriptor string (eg. `pkh(xpub661MyMwAqRbcFW31YEwpkMuc5THy2PSt5bDMsktWQcFF8syAmRUapSCGu8ED9W6oDMSgv6Zz8idoc4a6mr8BDzTJY47LJhkJ8UB7WEGuduB/2147483648`) the value of hard passed in the parsePath function will always be true since the values can be hardened as well as non-hardened. So the condition
 ` if (!hard && (index & common.HARDENED))
      throw new Error('Path index cannot be hardened.');` will never be checked when hard is true. 
The key path in the example descriptor (2147483648) is out of range. Without this condition it will not throw an error for these cases.